### PR TITLE
Allow to change intermediate events to boundary events

### LINKED ISSUE #478: Allow to change Intermediate Events to Boundary Events
* [x] Attach an intermediate event to a task to change it to a boundary event by dropping onto a task / sub-process border.
* [x] only works for some of catching and unspecified event definitions
* [ ] Keeps outgoing connections (if allowed)

### PR DESCRIPTION
**Pull request by barmac**
Closes #478

Builds on top of and includes changes in https://github.com/bpmn-io/bpmn-js/pull/1047

Requires https://github.com/bpmn-io/diagram-js/pull/354

### CODE REVIEW

**Comment by barmac** [comment]
Got stuck on layout:

When the event is replaced before the movement:
preExecute

When the event is replaced after the movement:
postExecuted

Goal:
Screenshot 2019-05-23 at 08 46 18

**Comment by nikku** [comment]
:heart_eyes:

**Comment by barmac** [comment]
This is not ready for (Parallel) Multiple Intermediate Catching Event 🙄

**Comment by barmac** [comment]
@nikku SubProcess behaves correctly, because expansion conducted in `toggleCollapse` command triggers ResizeShapeHandler which... (you guessed it) lays out the connections. I'm going to restore the flag.

**Comment by barmac** [comment]
@nikku Ready for review.

File: lib/util/model/ModelCloneUtils.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ import {
  * @type {Array}
  */
 export var IGNORED_PROPERTIES = [
+  'host',
```
Comment (nikku) [inline lib/util/model/ModelCloneUtils.js]: What is this about?

Comment (barmac) [reply lib/util/model/ModelCloneUtils.js]: I thought it might be useful in the context of host update, but actually we never use this. I'll remove it.

**Comment by falko** [comment]
Thank you!

**Pull request merged**